src/number.js

```diff
@@ -5,7 +5,7 @@ import isAbsent from './util/isAbsent';
 
 let isNaN = value => value != +value
 
-let isInteger = val => isAbsent(val) || val === (val | 0)
+let isInteger = val => isAbsent(val) || val === parseInt(val, 10)
```

Comment:
`parseInt` isn't the right thing, here we don't want to coerce the value into a number. The reason the original code doesn't work with large ints, btw, is that JS really only has 32bit ints and the bitwise operators first convert the float to an int then truncate it. This leads to overflows in the large number case. 

I'm a bit torn, the original method is more "correct" even if it does seemingly odd things. If we were gonna change it tho, we should use the `Number.isInteger` logic e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger#Polyfill

Reply:
I think the coercion is fine if strict equality is being used (`===`)

Reply:
But any approach that makes false negatives go away is fine by me 👍

Reply:
+1 for @jozanza approach, `parseInt` + strict equality FTW

Reply:
I finally realized the flaw of the `parseInt` approach:
```js
9999999999999999999.9 === parseInt(9999999999999999999, 10)
// true
```

I work on adding the `Number.isInteger` logic to this PR when I get a chance @jquense